src/Keboola/DbExtractor/Configuration/ConfigDefinitionValidationHelper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Configuration;
+
+class ConfigDefinitionValidationHelper
+{
+    public const MESSAGE_TABLE_OR_QUERY_MUST_BE_DEFINED = 'Either "table" or "query" must be defined.';
+    public const MESSAGE_TABLE_AND_QUERY_CANNOT_BE_SET_TOGETHER = 'Both "table" and "query" cannot be set together.';
+    public const MESSAGE_CUSTOM_QUERY_CANNOT_BE_FETCHED_INCREMENTALLY
+        = 'Incremental fetching is not supported for advanced queries.';
+
+
+    public static function isQueryAndTableNotDefined(array $table): bool
```

Comment:
isNeitherQueryNorTableDefined - tak jak je to teď to nedává IMHO smysl - resp je to $queryDefined && !$tableDefined

Reply:
https://dictionary.cambridge.org/grammar/british-grammar/questions-and-negative-sentences/neither-neither-nor-and-not-either

Reply:
A ještě "for the sake of completeness" https://dictionary.cambridge.org/grammar/british-grammar/linking-words-and-expressions/either-or :)

Reply:
Případně pojmenovat pozitivně "either X or Y defined" a pak to otočit v podmínce při použití. Logicky by to bylo čistější, ale v kódu je to čistější takhle myslím...

Reply:
`Neither` je to spravne slovo! S tim jak si to napsal jsem si pohraval, ale napadalo me tam strkat neco ve stylu `not`. Na neither jsem uplne zapomnel v tu chvili.

Reply:
mho je `isNeitherQueryNorTableDefined` je dobry

Reply:
> There are only two hard things in Computer Science: cache invalidation and naming things